x-pack/elastic-agent/pkg/agent/application/pipeline/dispatcher/dispatcher.go

```diff
@@ -74,7 +76,22 @@ func (ad *ActionDispatcher) key(a fleetapi.Action) string {
 }
 
 // Dispatch dispatches an action using pre-registered set of handlers.
-func (ad *ActionDispatcher) Dispatch(acker store.FleetAcker, actions ...fleetapi.Action) error {
+// ctx is used here ONLY to carry the span, for cancelation use the cancel
+// function of the ActionDispatcher.ctx.
+func (ad *ActionDispatcher) Dispatch(ctx context.Context, acker store.FleetAcker, actions ...fleetapi.Action) (err error) {
+	span, ctx := apm.StartSpan(ctx, "dispatch", "app.internal")
+	// Creating a child context that carries both the ad.ctx cancelation and
+	// the span from ctx.
+	var cancel func()
+	ctx, cancel = context.WithCancel(ad.ctx)
```

Comment:
If you make changes for the comment above, could you also change here to use `:=`?
Is there a special reason you avoid using `:=`?
```suggestion
	ctx, cancel := context.WithCancel(ad.ctx)
```

Reply:
no reason, adding it back